slave/brutedict/brutedict.go

```diff
@@ -0,0 +1,53 @@
+package brutedict
+
+import (
+	"math/big"
+
+	"github.com/aaronang/cong-the-ripper/lib"
+)
+
+type BruteDict struct {
+	task  *lib.Task
+	queue chan []byte
+	quit  chan bool
+}
+
+func New(task *lib.Task) (bd *BruteDict) {
+	bd = &BruteDict{
+		task:  task,
+		queue: make(chan []byte),
+		quit:  make(chan bool),
+	}
+
+	go bd.list(task.Alphabet, task.Start, task.TaskLen)
+	return
+}
+
+func (bd *BruteDict) Next() (candidate []byte) {
+	select {
+	case candidate = <-bd.queue:
+	case <-bd.quit:
```

Comment:
If `bd.quit` isn't there and `bd.queue` is empty then select will block. `bd.quit` will be filled when `list` is done, so that `Next` can return nil to inform the caller that it has finished.